internal/acceptance_test/resource_host_test.go

```diff
@@ -53,17 +66,92 @@ resource "hpegl_metal_host" "test_host" {
   machine_size       = "${data.hpegl_metal_available_resources.compute.machine_sizes.0.name}"
   ssh                = ["User1 - Linux"]
   networks           = sort(local.networks)
+  network_route      = "Public"
   location           = var.location
   description        = "hello from Terraform"
 }
 `
 }
 
-func testWaitUntilHostCreated() resource.TestCheckFunc {
+// testAccHostUpdateConfig updates the terraform config by updating
+// attributes of the host 'test_host':
+//   - description is updated
+//   - last network removed from sorted list of networks
+// Updated config is compared against config specified in testAccCheckHostBasic.
+func testAccHostUpdateConfig() string {
+	return `
+provider "hpegl" {
+	metal {
+	}
+}
+
+variable "location" {
+	default = "USA:Central:V2DCC01"
+}
+
+data "hpegl_metal_available_resources" "compute" {
+}
+
+locals  {
+	host_os_flavor = "${data.hpegl_metal_available_resources.compute.images.0.flavor}"
+	host_os_version = "${data.hpegl_metal_available_resources.compute.images.0.version}"
+	networks = ([for net in "${data.hpegl_metal_available_resources.compute.networks}": 
+		net.name if net.location == var.location] )
+	sorted_networks = sort(local.networks)
+	updated_networks_length = length(local.sorted_networks) - 1
+	updated_networks = ([for i, net in local.sorted_networks : net if i < local.updated_networks_length])
+}
+
+resource "hpegl_metal_host" "test_host" {
+	name               = "test"
+	image              = join("@",[local.host_os_flavor, local.host_os_version])
+	machine_size       = "${data.hpegl_metal_available_resources.compute.machine_sizes.0.name}"
+	ssh                = ["User1 - Linux"]
+	networks           = local.updated_networks
+	network_route      = "Public"
+	location           = var.location
+	description        = "hello from Terraform (updated)"
+}
+`
+}
+
+// testWaitUntiHostReady checks if the host was created successfully and
+// is in the 'Ready' state.
+func testWaitUntiHostReady(rsrc string) resource.TestCheckFunc {
 	return func(s *terraform.State) error {
-		// TODO
-		// 1. Poll the state instead of fixed wait time
-		time.Sleep(hostCreateWait)
+		rs, ok := s.RootModule().Resources[rsrc]
+		if !ok {
+			return fmt.Errorf("Host not found: %q", rsrc)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No host primary ID set")
+		}
+
+		hostID := rs.Primary.ID
+
+		p, err := client.GetClientFromMetaMap(testAccProvider.Meta())
+		if err != nil {
+			return fmt.Errorf("Error retrieving Metal client: %v", err)
+		}
+
+		ctx := p.GetContext()
+
+		hostState := rest.HOSTSTATE_NEW
+		for i := 0; i < hostStatePollCount && hostState != rest.HOSTSTATE_READY; i++ {
+			host, _, err := p.Client.HostsApi.GetByID(ctx, hostID)
+			if err != nil {
+				return fmt.Errorf("Host: %q not found: %s", hostID, err)
+			}
+
+			time.Sleep(hostStateReadyWait)
```

Comment:
Sleep should be moved to beginning of loop.  Otherwise, you waste a sleep cycle if the HostsApi.GetByID() returns the host already in the desired ready state before checking that it is ready.